### PR TITLE
Rerender on hydrate / add static content component

### PR DESCRIPTION
**Pull request by NicoZweifel**
## Summary
This ~~is a work in progress and~~ aims at improving the dev experience regarding hydration errors.
The hydration errors that are happening in dev are caused by the replacing of the children in the tree in `getStaticProps/getMdxSource`, especially `withSyntaxHighlighting`:

```tsx
withCodeBlocks,
        [
          withSyntaxHighlighting,
          {
            ignoreMissing: true, // Ignore errors when no language is found
          },
        ],
```

This PR adds a `StaticContent` Component that can be wrapped around code elements that should be static or have hydration issues to either analyze and fix the issue or disable errors/warnings in dev.
It will show false positives and some weird mismatches because of how the component works, this can be improved for sure. 

But for a lot of hydration errors, especially when testing on a Page with an individual component it can really help:

image

It will only work on the Main Content (i.e. components/callouts) and not on the SidePanel.
The SidePanel `CodeGroup` causes an hydrationError which is caused by Prism, but it has to be silenced differently because there is no MDX there (yet):
```tsx
<code
    dangerouslySetInnerHTML={{
      __html: Prism.highlight(
        snippet.code,
        snippet.prism.grammar,
        snippet.prism.language
      ),
    }}
  />
```

## Test Plan

Tested so far locally with indvidual docs. ~~Will improve and refactor this PR still and move/cleanup files.~~

Would be happy to get another opinion on this, feel free to suggest improvements  and ideas. ~~but keep in mind that this is work in progress.~~

Will only affect dev environment. In prod the minified errors will still be in the console but no hydration popup in dev and warnings in the console instead.

## Possible Improvements

This is not ideal and is sort of a last resort at removing hydration errors.
We can hopefully remove this when moving to `appDir` and moving away from getStaticProps with `next` 13.

### CODE REVIEW

**Review by tobias-mintlify — APPROVED** [review]
+1 that we might not need this when we move to the app dir, but this looks great in the mean time!

File: client/src/components/Static/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { StaticCode } from './StaticCode';
+import { StaticContent } from './StaticContent';
+import { StaticPre } from './StaticPre';
+import { StaticSpan } from './StaticSpan';
+
+export { StaticContent, StaticCode, StaticPre, StaticSpan };
+export const staticComponents: MDXComponents = {
+  pre: StaticPre,
+  code: StaticCode,
+  span: StaticSpan,
```
[the remark this answers is not in the document]
Comment (NicoZweifel) [reply client/src/components/Static/index.ts:12]: Yea exactly

File: client/src/components/Static/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { MDXComponents } from 'mdx/types';
```
[the remark this answers is not in the document]
Comment (NicoZweifel) [reply client/src/components/Static/index.ts:1]: Added a comment with link to this PR to `Static/index.ts`

[the remark this answers is not in the document]
Comment (hahnbeelee) [reply client/src/components/Static/index.ts:1]: amazing

**Pull request merged**